src/stactools/landsat/utils.py

```diff
@@ -98,3 +98,34 @@ def handle_antimeridian(item: Item, antimeridian_strategy: Strategy) -> None:
         item.geometry = Polygon(merged_coords)
 
     antimeridian.fix_item(item, antimeridian_strategy)
+
+
+def round_coordinates(item: Item, precision: int = 6) -> None:
+    """Rounds an Item's geometry and bbox geographic coordinates.
+
+    Any tuples encountered will be converted to lists.
+
+    Args:
+        item (Item): pystac Item.
+        precision (int): Number of decimal places for rounding. Defaults to 6,
+            which is approximately 1 cm at the equator.
+    """
+
+    def recursive_round(coordinates: List[Any], precision: int) -> List[Any]:
```

Comment:
```suggestion
    def recursive_round(coordinates: Iterable[Any], precision: int) -> List[float]:
```

Reply:
- `coordinates` actually needs to be a `List`, not a generic `Iterable`. For example, it cannot be a Tuple (that's pronounced _tupple_, mind you), because it would not be able to be updated in place. Note the conversion to list for both geometry and bbox before the call.
- The return type can be a List, or a few different structures of nested Lists, so `List[Any]` is more appropriate, no?

Reply:
> coordinates actually needs to be a List, not a generic Iterable. For example, it cannot be a Tuple (that's pronounced tupple, mind you), because it would not be able to be updated in place. Note the conversion to list for both geometry and bbox before the call.

Ah yeah, I guess I didn't think about the implications of the in-place modifications. Disregard.

>The return type can be a List, or a few different structures of nested Lists, so List[Any] is more appropriate, no?

Correct, my mistake, disregard.